webapp/src/index.js

```diff
@@ -16,12 +17,24 @@ class Plugin {
     // eslint-disable-next-line no-unused-vars
     initialize(registry, store) {
         registry.registerChannelHeaderButtonAction(
-            <Icon/>,
+            <ChannelHeaderIcon/>,
             (channel) => {
                 startMeeting(channel.id)(store.dispatch, store.getState);
             },
             'Start Zoom Meeting',
+            'Start Zoom Meeting',
         );
+
+        if ('registerAppBarComponent' in registry) {
```

Comment:
Does the webapp PR contain the logic that hides the `ChannelHeaderButton` if the AppBar is enabled?

Reply:
Please see the description of this zoom PR that explains how this works

Reply:
@hanzei

Reply:
But yes you're right: If the app bar is disabled, then all channel header icons will unconditionally be displayed

Reply:
For these plugins that we are registering in both places, they will show in both, but what about if we only register the plugin on the App Bar?

Reply:
> For these plugins that we are registering in both places, they will show in both

@larkox In my active branch on the webapp, I'm making it so if a plugin registers components in both locations, and the app bar is enabled, then we won't show the channel header components for that plugin.

> but what about if we only register the plugin on the App Bar?

Since the app bar can be toggled through feature flag, the webapp needs to know about the channel header components so it can show those when appropriate.